doc/spec.md

```diff
@@ -2348,7 +2355,62 @@ foo({ id: 1234, name: false });    // Error, name of wrong type
 foo({ name: "hello" });            // Error, id required but missing
 ```
 
-### <a name="3.11.5"/>3.11.5 Excess Properties
+### <a name="3.11.5"/>3.11.5 Comparability
+
+*TODO: Document the base primitive type.
+
+*Types are required to be *comparable* in certain circumstances, such as part of when checking whether two values of given types might be equal at runtime using operators like '===', or when using a type assertion.
+
+*S* is ***comparable to*** a type *T*, and *T* is ***comparable from*** *S*, if one of the following is true:
+
+* *S* and *T* are identical types.
+* *S* or *T* is the Any type.
```

Comment:
`simpleTypeRelatedTo` rules go here, overwriting lines 2367-2369. (Note that `simpleTypeRelatedTo` is called twice, but I flatten the rules so we don't have to say "only this part is symmetric".)

* *S* is the never type.
* *S* is a literal type and *T* is the base primitive type of *S*. (note: and vice versa for comparability)
* *S* is an enum member with a literal type and *T* is a literal type that is not an enum member, and the literal values of *S* and *T* are the same. (note: and vice versa, for comparability)
* *S* and *T* are enum members and come from the same enum.
* *S* and *T* are enum members with a literal type and the literal values of *S* and *T* are the same, and *S* and *T* come from the same enum.
* *S* and *T* are the undefined or void types. (note: for assignability, the rule is "*S* is the undefined type and *T* is the undefined or void types.")
* *S* and *T* are the null type.
* *S* or *T* is an object type and the other is the `object` type. (note: assignability is "*S* is an object type and *T* is the `object` type.")
* *S* or *T* is an enum type and the other is the number type or a numeric literal type that is not the type of an enum member.

Reply:
We need a section that describes how enums "come from the same enum" - because we really need an enum relation section.

Reply:
Also "*S* and *T* are enum types and come from the same enum" instead of "*S* and *T* are enum members ...".